.github/workflows/cicd.yml

```diff
@@ -0,0 +1,248 @@
+        uses: actions/checkout@v1
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run Mypy
+        run: tox -e mypy;
+
+
+  docs:
+
+    name: Build documentation
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v1
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Build documentation
+        run: tox -e docs;
+
+
+  docs-linkcheck:
+
+    name: Documentation link check
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v1
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Check for broken links in documentation
+        run: tox -e docs-linkcheck;
+
+
+  packaging:
+    name: Packaging
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v1
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Check packaging
+        run: tox -e packaging;
+
+
+  unit:
+    name: Unit Tests using Python ${{ matrix.python }} on Ubuntu with Twisted ${{ matrix.twisted }}
+
+    needs: [flake8, black, mypy, docs, packaging]
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        python: ["2.7", "3.5", "3.6", "3.7", "3.8", "pypy2", "pypy3"]
+        twisted: ["17.1", "18.4", "current"]
+
+    steps:
+
+      - name: Checkout source code
+        uses: actions/checkout@v1
+
+      - name: Install Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install Tox
+        run: pip install tox;
+
+      - name: Run unit tests
+        shell: bash
+        run: |
+          py="${{ matrix.python }}";
+          if [ "${py#pypy}" != "${py}" ]; then  # PyPy
+              py_test="${py}";
+              py_codecov="${py}";
+          else  # CPython
+              py_test="py${py/./}";  # Add "py" prefix, remove "."
+              py_codecov="py";
+          fi;
+          tw="${{ matrix.twisted }}";
+          tw="tw${tw/./}";  # Remove "."
+          env_test="coverage-${py_test}-${tw}";
+          env_codecov="codecov-${py_codecov}";
+          ln -s ".tox/${env_test}" testenv;  # For upload step
+          tox -e "${env_test},${env_codecov}";
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
(Once you enable this for PRs you'll need to make this public FWIW)

Reply:
Yeah GHA doesn't inject secrets into forks. You should use the official plugin: https://hynek.me/articles/python-github-actions/